### ISSUE TITLE
Any plan on introducing the ext_session_lock_manager_v1 protocol?

### ISSUE THREAD

**Issue by ntrp** [riverwm/river#602]
I have seen it is implemented by `waylock` since I get this error:

```
error: ext_session_lock_manager_v1 not advertised

error: failed to read password from pipe: EndOfStream
```

I am trying to setup gpg key auto unlock with `pam-gnupg` and currently there is no wayland screen locker calling `pam_setcred` after unlock, not even `swaylock`

**Comment by novakne** [comment]
You need #497 and get the wlroots branch mentionned in the PR to use it, as `ext_session_lock` is not yet in a wlroots stable release

**Comment by ifreund** [comment]
Yes, there are plans and the work is already done in #497. We're just waiting on a wlroots release at this point before that PR can be merged. You're welcome to compile river from source with that patch and the linked wlroots branch until then.

**Comment by ifreund** [comment]
> I am trying to setup gpg key auto unlock with `pam-gnupg` and currently there is no wayland screen locker calling `pam_setcred` after unlock, not even `swaylock`

swaylock does seem to call `pam_setcred()` but it passes `PAM_REFRESH_CRED` instead of `PAM_REINITIALIZE_CRED` like waylock's master branch does. Perhaps that difference is what's causing your issues?
